### PR TITLE
Improve streaming sync routes

### PR DESCRIPTION
**Pull request by dholms**
Previously, we were reading the car file to a buffer before writing it out to the response.

Now we just stream it directly out

### CODE REVIEW

File: packages/common/src/streams.ts [hunk 1]
```diff
@@ -38,6 +38,24 @@ export const streamToBytes = async (stream: Readable): Promise<Uint8Array> => {
   return new Uint8Array(Buffer.concat(bufs))
 }
 
+const pushIterableToStream = async (
+  iter: AsyncIterable<Uint8Array>,
+  stream: Readable,
+) => {
+  for await (const bytes of iter) {
+    stream.push(bytes)
+  }
```
Comment (devinivy) [inline packages/common/src/streams.ts:47]: If this throws, I think we should ensure `stream` gets destroyed here.

Comment (devinivy) [reply packages/common/src/streams.ts:47]: Actually, `Readable.from()` may even do all this for us.

File: packages/pds/src/sql-repo-storage.ts [hunk 2]
```diff
@@ -167,6 +167,7 @@ export class SqlRepoStorage extends RepoStorage {
         size: bytes.length,
         content: bytes,
       })
+      this.cache.set(cid, bytes)
```
Comment (devinivy) [inline packages/pds/src/sql-repo-storage.ts:170]: Just making sure this was an intentional addition!

**Review by devinivy — APPROVED** [review]
This rocks 💎

File: packages/common/src/streams.ts (hunk 1 above)
Comment (dholms) [reply packages/common/src/streams.ts:47]: wow nice i searched around for that exact fn & didn't find it 🙃

**Review by devinivy — APPROVED** [review]

**Pull request merged**